src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -194,6 +208,126 @@ public async Task ExecuteAsync(string stackName, bool saveCdkProject)
             await orchestrator.DeployRecommendation(cloudApplication, selectedRecommendation);
         }
 
+        /// <summary>
+        /// This method is used to set the values for Option Setting Items when a deployment is being performed using a user specifed config file.
+        /// </summary>
+        /// <param name="recommendation">The selected recommendation settings used for deployment <see cref="Recommendation"/></param>
+        /// <param name="deploymentBundleDefinition">The container for the deployment bundle used by an application. <see cref="DeploymentBundleDefinition"/></param>
+        /// <param name="userDeploymentSettings">The deserialized object from the user provided config file. <see cref="UserDeploymentSettings"></param>
+        private void ConfigureDeployment(Recommendation recommendation, DeploymentBundleDefinition deploymentBundleDefinition, UserDeploymentSettings userDeploymentSettings)
+        {
+            foreach (var entry in userDeploymentSettings.LeafOptionSettingItems)
+            {
+                var optionSettingJsonPath = entry.Key;
+                var optionSettingValue = entry.Value;
+
+                var isPartOfDeploymentBundle = true;
+                var optionSetting = deploymentBundleDefinition.Parameters.FirstOrDefault(x => x.Id.Equals(optionSettingJsonPath));
+                if (optionSetting == null)
+                {
+                    optionSetting = recommendation.GetOptionSetting(optionSettingJsonPath);
+                    isPartOfDeploymentBundle = false;
+                }

```

Comment:
I don't like what you are doing here. We don't need to differentiate between the deployment bundle option settings and the rest of the option settings. At the end, they are all option settings with Type Hints. As long as we are executing the type hints, we shouldn't have an issue. You don't need to manually execute the type hint commands based on a switch.

Reply:
There are 2 issues with trying to invoke TypeHintCommands for **_non-deployment-bundle option settings._**
* The TypeHintCommands are only available for top level option settings (ex. IAMRole, ECSCluster, VPC etc). However, when performing a deployment using a config file, we are only dealing with **_leaf option settings_** (ex. IAMRole.CreateNew, ECSCluster.NewClusterName etc.)
* These TypeHintCommands are only responsible for reading the input from the CLI and create a corresponding TypeHinResponse based on those values. But since we are only dealing **_leaf option settings_**, we can directly set its value by calling in the `SetValueOverride` method. We will lose separation of concern if we amend the existing TypeHintCommands to facilitate the deployment using a config file (and will also complicate things in the process)

For the **_deployment-bundle option settings_**, the TypeHintCommands also perform some validation (since there are no validators specified as part of the recipe file). Due to this dependency I added the `OverrideValue` method that takes in the desired value, performs the **same validations** on it and finally sets it as part of the deployment bundle.